service/matching/db_task_queue_ownership.go

```diff
@@ -92,6 +104,7 @@ func newDBTaskQueueOwnership(
 	store persistence.TaskManager,
 	logger log.Logger,
 ) *dbTaskQueueOwnershipImpl {
+	name, _ := newTaskQueueName(taskQueueKey.TaskQueueName)
```

Comment:
should not ignore error. Maybe you should require the input to be of type taskQueueID